internal/mode/static/handler.go

```diff
@@ -92,3 +114,42 @@ func (h *eventHandlerImpl) updateNginx(ctx context.Context, conf dataplane.Confi
 
 	return nil
 }
+
+// updateControlPlaneAndSetStatus updates the control plane configuration and then sets the status
+// based on the outcome
+func (h *eventHandlerImpl) updateControlPlaneAndSetStatus(ctx context.Context, cfg *nkgAPI.NginxGateway) {
+	var cond []conditions.Condition
+	if err := updateControlPlane(
```

Comment:
Right now the function returns an error when there's a problem with processing the configuration. The event case isn't technically an error, but more of a warning that we're reverting to defaults since a config doesn't exist. So this makes it a bit less clean to return, because then we have to have the caller figure out if that error is actually an error or just a warning. I think it's easier to just have this function contain that logic, and the caller always knows that and error is an error.

Reply:
> The event case isn't technically an error, but more of a warning that we're reverting to defaults since a config doesn't exist

there is a bit of mismatch here 🤔 . because if the updateControlPlane, we have:
```
		msg := "NginxGateway configuration was deleted; using defaults"
		logger.Error(nil, msg)
```
so we log it as an error.

Reply:
That's true...we don't have warning log levels right now. We could either add that support here, or log it at info. What do you think?

Reply:
per guidelines, let's log it as info https://github.com/nginxinc/nginx-kubernetes-gateway/blob/main/docs/developer/logging-guidelines.md#when-to-log

Reply:
Getting good use out of your new doc ;)